### PR TITLE
added-surveysparrow-source-and-destination-doc

### PR DESCRIPTION
**Pull request by surveysparrowuser**
added SurveySparrow source and destination document

### CODE REVIEW

File: src/connections/destinations/catalog/surveysparrow/index.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
+Title: SurveySparrow (Actions) Destination
+---
+{% include content/plan-grid.md name="actions" %}
+
+[SurveySparrow](https://surveysparrow.com/?utm_source=segmentio&utm_medium=docs&utm_campaign=partners){:target="_blank"} is an end to end omnichannel experience management platform that bundles Customer Experience and Employee Experience tools such as NPS, Offline, Chat, Classic, and 360 Surveys which are mobile-first, highly engaging, and user-friendly.
```
Comment (cmastr) [inline src/connections/destinations/catalog/surveysparrow/index.md:5]:
```suggestion
[SurveySparrow](https://surveysparrow.com/?utm_source=segmentio&utm_medium=docs&utm_campaign=partners){:target="_blank"} is an end-to-end omnichannel experience management platform that bundles Customer Experience and Employee Experience tools such as NPS, Offline, Chat, Classic, and 360 Surveys which are mobile-first, highly engaging, and user-friendly. 
```

File: src/connections/destinations/catalog/surveysparrow/index.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+[SurveySparrow](https://surveysparrow.com/?utm_source=segmentio&utm_medium=docs&utm_campaign=partners){:target="_blank"} is an end to end omnichannel experience management platform that bundles Customer Experience and Employee Experience tools such as NPS, Offline, Chat, Classic, and 360 Surveys which are mobile-first, highly engaging, and user-friendly. 
+
+This destination is maintained by SurveySparrow. For any issues with the destination, [contact their Support team](mailto:support@surveysparrow.com).
+
+## Getting started
+
+1. From your workspace's [Destination catalog page](https://app.segment.com/goto-my-workspace/destinations/catalog){:target="_blank"} search for "SurveySparrow"
+2. Select SurveySparrow and click **Add Destination**
+3. Select an existing Source to connect to SurveySparrow (Actions).
+4. Login to your [SurveySparrow](https://app.surveysparrow.com/) account, navigate to **Settings > Apps and Integrations > Create a Custom app**.
```
Comment (cmastr) [inline src/connections/destinations/catalog/surveysparrow/index.md:14]:
```suggestion
4. Log in to your [SurveySparrow](https://app.surveysparrow.com/) account, then navigate to **Settings > Apps and Integrations > Create a Custom app**.
```

File: src/connections/destinations/catalog/surveysparrow/index.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+5. Fill in the details for custom app and choose **Select all** under Scope. Later, you can remove the scopes which are not required.
```
Comment (cmastr) [inline src/connections/destinations/catalog/surveysparrow/index.md:15]:
```suggestion
5. Fill in the details for the custom app and choose **Select all** under Scope. Later, you can remove any scopes that are not required.
```

File: src/connections/sources/catalog/cloud-apps/surveysparrow/index.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
+title: SurveySparrow Source
+---
+[SurveySparrow](https://surveysparrow.com/?utm_source=segmentio&utm_medium=docs&utm_campaign=partners){:target="_blank"} is an end to end omnichannel experience management platform that bundles Customer Experience and Employee Experience tools such as NPS, Offline, Chat, Classic, and 360 Surveys which are mobile-first, highly engaging, and user-friendly.
```
Comment (cmastr) [inline src/connections/sources/catalog/cloud-apps/surveysparrow/index.md:3]:
```suggestion
[SurveySparrow](https://surveysparrow.com/?utm_source=segmentio&utm_medium=docs&utm_campaign=partners){:target="_blank"} is an end-to-end omnichannel experience management platform that bundles Customer Experience and Employee Experience tools such as NPS, Offline, Chat, Classic, and 360 Surveys which are mobile-first, highly engaging, and user-friendly. 
```

File: src/connections/sources/catalog/cloud-apps/surveysparrow/index.md [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+This is an [Event Cloud Source](/docs/sources/#event-cloud-sources) which can not only export data into your Segment warehouse, but can also federate the exported data into your other enabled Segment Destinations.
+
+This source is maintained by SurveySparrow. For any issues with the source, [contact their Support team](mailto:support@surveysparrow.com).
+
+## Getting started
+
+1. From your workspace's [Sources catalog page](https://app.segment.com/goto-my-workspace/sources/catalog){:target="_blank"} click **Add Source**.
+2. Search for "SurveySparrow" in the Sources Catalog, select SurveySparrow, and click **Add Source**.
+3. On the next screen, give the Source a name configure any other settings.
```
Comment (cmastr) [inline src/connections/sources/catalog/cloud-apps/surveysparrow/index.md:13]:
```suggestion
3. On the next screen, give the Source a name, then configure any other settings.
```

File: src/connections/sources/catalog/cloud-apps/surveysparrow/index.md [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+4. Click **Add Source** to save your settings.
+5. Copy the Write key from the Segment UI.
+6. Log in to your SurveySparrow account - navigate to Settings > Apps and Integrations > Segment Source Integration and paste the Write key and select the Segment data processing region to connect.
+
+## Stream
+
+SurveySparrow uses our stream Source component to send Segment event data. It uses a server-side (`track`) method to send data to Segment. These events are then available in any destination that accepts server-side events, and available in a schema in your data warehouse, so you can query using SQL.
+
+The default behavior is for SurveySparrow to pass the hash value of email and mobile number of respondent as the anonymousId. There are cases in which SurveySparrow does not have an associated email or mobile number, in which case a random hash will be passed in as the anonymousId.
```
Comment (cmastr) [inline src/connections/sources/catalog/cloud-apps/surveysparrow/index.md:25]:
```suggestion
The default behavior is for SurveySparrow to pass the hash value of the email address and mobile number of the respondent as the `anonymousId`. There are cases in which SurveySparrow does not have an associated email address or mobile number, in which case a random hash will be passed in as the `anonymousId`.
```

File: src/connections/sources/catalog/cloud-apps/surveysparrow/index.md [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+
+The table below lists events that SurveySparrow sends to Segment. These events appear as tables in your warehouse, and as regular events in other Destinations. SurveySparrow includes the `anonymousId` if available.
+
+| Event Name         | Description                           |
+| ------------------ | ------------------------------------- |
+| Submission Completed  | Survey response was submitted successfully         |
+
+## Event Properties
+
+The table below list the properties included in the events listed above.
```
Comment (cmastr) [inline src/connections/sources/catalog/cloud-apps/surveysparrow/index.md:37]:
```suggestion
The table below lists the properties included in the events listed above.
```

File: src/connections/sources/catalog/cloud-apps/surveysparrow/index.md [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+| `contact`       | Contact which has created the response         |
+| `expressions`   | Computed expressions data of the survey         |
+| `variables`     | Computed variables data of the survey         |
+| `answers`       | contains answer details of the entire survey         |
+
+## Adding Destinations
+
+Now that your Source is set up, you can connect it with Destinations.
+
+Log into your downstream tools and check to see that your events appear as expected, and that they contain all of the properties you expect. If your events and properties don't appear, check the [Event Delivery](/docs/connections/event-delivery/) tool, and refer to the Destination docs for each tool for troubleshooting.
```
Comment (cmastr) [inline src/connections/sources/catalog/cloud-apps/surveysparrow/index.md:53]:
```suggestion
Log in to your downstream tools and check to see that your events appear as expected, and that they contain all of the properties you expect. If your events and properties don't appear, check the [Event Delivery](/docs/connections/event-delivery/) tool, and refer to the Destination docs for each tool for troubleshooting.
```

**Review by cmastr — APPROVED** [review]

File: src/connections/destinations/catalog/surveysparrow/index.md [hunk 9]
```diff
@@ -0,0 +1,44 @@
+Title: SurveySparrow (Actions) Destination
+---
```
Comment (tcgilbert) [inline src/connections/destinations/catalog/surveysparrow/index.md:2]:
```suggestion
---
title: SurveySparrow (Actions) Destination
hidden: true
beta: true
---
```

File: src/connections/destinations/catalog/surveysparrow/index.md [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+## Action destination for Contact creation event
+
+| Property Name    | Type          | Description                    |
+| ------------- |:-------------:| :-----                         |
+| `full_name`     | string        | Full Name of contact           |
+| `phone`         | string        | Phone number of the contact    |
+| `mobile`        | string        | Mobile number of the contact   |
+| `email         | string        | Email of contact               |
+| `job_title`     | string        | Job Title of the contact       |
+| `contact_type`  | string        | Type of contact [contact,employee] |
```
Comment (tcgilbert) [inline src/connections/destinations/catalog/surveysparrow/index.md:40]: Could you explain the use case for having these tables? It's likely you can remove this as `{% include components/actions-fields.html %}` will populate all the information about the Actions you created and its fields

File: src/connections/sources/catalog/cloud-apps/surveysparrow/index.md [hunk 11]
```diff
@@ -0,0 +1,55 @@
+title: SurveySparrow Source
+---
```
Comment (tcgilbert) [inline src/connections/sources/catalog/cloud-apps/surveysparrow/index.md:2]:
```suggestion
---
title: SurveySparrow Source
id: di2sOHoscX
hidden: true
beta: true
---
```

File: src/connections/sources/catalog/cloud-apps/surveysparrow/index.md [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+
+This source is maintained by SurveySparrow. For any issues with the source, [contact their Support team](mailto:support@surveysparrow.com).
+
+## Getting started
+
+1. From your workspace's [Sources catalog page](https://app.segment.com/goto-my-workspace/sources/catalog){:target="_blank"} click **Add Source**.
+2. Search for "SurveySparrow" in the Sources Catalog, select SurveySparrow, and click **Add Source**.
+3. On the next screen, give the Source a name, then configure any other settings.
+
+   - The name is used as a label in the Segment app, and Segment creates a related schema name in your warehouse. The name can be anything, but we recommend using something that reflects the source itself and distinguishes amongst your environments (eg. SourceName_Prod, SourceName_Staging, SourceName_Dev).
```
Comment (tcgilbert) [inline src/connections/sources/catalog/cloud-apps/surveysparrow/index.md:15]:
```suggestion
   - The name is used as a label in the Segment app, and Segment creates a related schema name in your warehouse. The name can be anything, but we recommend using something that reflects the source itself and distinguishes amongst your environments (eg. SurveySparrow_Prod, SurveySparrow_Staging, SurveySparrow_Dev).
```

**Review by tcgilbert — CHANGES_REQUESTED** [review]

**Review by tcgilbert — CHANGES_REQUESTED** [review]
@surveysparrowuser one more thing is that you will need to change the folder name for the destination: 

- current: `surveysparrow/index.md`
- change to: `actions-surveysparrow/index.md`

**Review by tcgilbert — APPROVED** [review]

**Pull request merged**